app/src/main/java/com/gh4a/utils/HttpImageGetter.java

```diff
@@ -534,7 +534,6 @@ private Bitmap renderSvgToBitmap(Resources res, InputStream is) {
         try {
             SVG svg = SVG.getFromInputStream(is);
             if (svg != null) {
-                svg.setRenderDPI(DisplayMetrics.DENSITY_DEFAULT);
```

Comment:
Please explain the exact problem. I do not remember details, but I *do* remember this line was part of a lot of fiddling around to get SVGs render in correct size in all cases.

Reply:
You can see it in the app with the following image, whose root `<svg>` element has a width specified in `pt`s:
![width_pt](https://user-images.githubusercontent.com/30041551/188101647-9192a893-f205-442e-a8de-a2cced0f2429.svg)

Setting the render DPI to a different value causes the above image to be scaled incorrectly. More specifically, sizes returned by `getDocumentWidth/Height` will be way higher than the actual sizes of the image (because `DENSITY_DEFAULT` is higher than the default value of 96 used by the library) and therefore you'll see a big margin around the image because the `density` variable used a few lines below gets divided by 2 more times than it's actually necessary.
This change doesn't affect SVGs whose size is defined in pixels, because the render DPI is only used to convert "real world" units (pts, cms, inches) in pixels.

Reply:
The thing is, rendering at 96dpi does not make it correct, since the pixel density of the device's screen is not `96 dpi * density`, but `160 dpi * density`. The basic idea is
- make the library render the image at `mdpi` (160 dpi)
- do the upscaling for higher densities via canvas scaling

Where does the default 96 dpi come into play here?

Reply:
For now I've reverted the change.
As it is now there is clearly some issue with the scaling, either with the DPI or with the canvas scaling. Removing the line makes the image look way more correct (just look at the comment above in the app) but I don't know if that's the most correct way to fix it.
By the way, currently the app does not scale raster images according to the screen DPI either.
My feeling is that we should use a library like Coil or Picasso to handle image loading and rendering instead of doing it manually... it's a burden and quite error prone.

Reply:
> can not be meaningfully scaled without quality loss

That's the price to pay when using high DPI displays ¯\_(ツ)_/¯... you can simply open the OctoDroid Readme from the Files list to see how the WebView properly scales raster images, compared to the Readme preview in the repository About tab.

Regarding libraries, [Coil](https://coil-kt.github.io/coil/) doesn't force you to load images in an ImageView: it can fetch and return them as Drawables after scaling/transforming them. It also has built-in support for SVGs backed by the same library we use.
Once minSdk is bumped to 21, it would be a viable way to simplify the code and avoid doing all the work manually.